### PR TITLE
docs(website): add remark plugin for tabs and apply it to rules

### PR DESCRIPTION
**Pull request by armano2**
## Overview

This pr ads easy way to use tabs component and apply it for examples in some of rules

while i was on it i fixed up some of examples, I'm going to apply this to all of rules, if this is preferred change

### How to use it

```
<!--tabs-->

## Tab 1

content

## Tab 2

content
```

### Example

web: [/rules/class-literal-property-style](https://deploy-preview-4128--typescript-eslint.netlify.app/rules/array-type)
source [/rules/class-literal-property-style](https://github.com/typescript-eslint/typescript-eslint/blob/docs/code-examples/packages/eslint-plugin/docs/rules/array-type.md)

image

### CODE REVIEW

File: packages/website/src/remark/tabs.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        })
+        .join('')
+        .trim();
+    }
+
+    let tabNodes = [];
+
+    tabNodes.push({
+      type: 'jsx',
+      value: `<Tabs defaultValue="${getId(tabs[0])}" values={[${tabs
```
Comment (Josh-Cena) [inline packages/website/src/remark/tabs.js:56]: In recent Docusaurus versions you don't need to use `defaultValue` and `values` in `Tabs` anymore (https://docusaurus.io/docs/markdown-features/tabs), so `defaultValue` will be inferred as the first tab, and you could just move the `label` prop to each `TabItem` node.

**Comment by Josh-Cena** [comment]
This looks very eye-opening, making the doc work in both GitHub Markdown viewer and on Docusaurus... Maybe we could mention this in the Docusaurus docs👀 Note that this may break in MDX 2.0 though, if they don't decide to support HTML comments

**Comment by bradzacher** [comment]
I like the compact style of this. I don't think it sacrifices any readability.
Might be good to put some emoji in the tab header to make them clearer as well like

❌ Incorrect ✅ Correct

**Comment by armano2** [comment]
emoji added, and i applied this to more rules (half done)

**Comment by armano2** [comment]
one more small update tab heading now are going to be excluded from toc (this list on right)

**Comment by armano2** [comment]
@Josh-Cena i did some refactoring of this plugin, it should now generate simpler code

**Comment by Josh-Cena** [comment]
Yes, LGTM👍

File: packages/website/src/remark/tabs.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+      root.children.unshift({
+        type: 'import',
+        value: "import Tabs from '@theme/Tabs';",
+      });
+    }
+  };
+}
+
+module.exports = tabs;
```
Comment (JoshuaKGoldberg) [inline packages/website/src/remark/tabs.js:120]: This is really cool, TIL how to set up a plugin like this. Thanks! 😄

File: packages/website/src/remark/tabs.license.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+original code https://github.com/mccleanp/remark-docusaurus-tabs
```
Comment (JoshuaKGoldberg) [inline packages/website/src/remark/tabs.license.md:1]: Why not use the original code directly? Should we file an issue to look at making a fork / helping take ownership / etc?

**Review by JoshuaKGoldberg — DISMISSED** [review]
Love it!

File: packages/website/src/remark/tabs.license.md (hunk 3 above)
Comment (armano2) [reply packages/website/src/remark/tabs.license.md:1]: i should actually make pr in there

Comment (armano2) [reply packages/website/src/remark/tabs.license.md:1]: i did push pr to this package, https://github.com/mccleanp/remark-docusaurus-tabs/pull/2

**Review by bradzacher — DISMISSED** [review]

**Comment by bradzacher** [comment]
just checking @armano2 - is this ready? i see there's only 39 files modified - did we need to do the rest of the rules to match it?

**Comment by armano2** [comment]
yes we should update rest of rules i think

**Comment by armano2** [comment]
> 
> 
> just checking @armano2 - is this ready? i see there's only 39 files modified - did we need to do the rest of the rules to match it?

ok i updated all remaining rules

**Review by bradzacher — APPROVED** [review]
sweeet

**Pull request merged**